### DISCUSSION TITLE
Minutes

### DISCUSSION THREAD

**Discussion by IllustratedMan-code** [IllustratedMan-code/SeniorProject#9] [General]
Here is the minutes

**Comment by IllustratedMan-code** [comment]
- Set up discussions for the minutes

**Reply by IllustratedMan-code** [reply]
Changing the repo to its own organization broke the minutes

**Comment by IllustratedMan-code** [comment]
- Larger game with minigames to unlock traits?
- Take other genetic's related games and simplify them as minigames
- Different "worlds" that exercise different concepts
- Angry birds
- Predator prey
- mouse game (bullet hell)
- sickle cell anemia

**Comment by adediu25** [comment]
- 1920x1080 res, 16:9 aspect ratio as standard
- Adding scenes into repo may break them
- Alex continuing developing predator/prey tower defense style game, create write-up for game idea/content
- Connor start work on build the codons mini game
- David can be in charge of assets/art style
- John looking at entity component system for the week
- David continuing work on mutation game
- Module 7 essay completed, review and edit if necessary

**Comment by adediu25** [comment]
- Alex demo TD mini game, have to think of good lesson(s) for minigame
- Alex to investigate developing random game sounds (game over, clicking, etc)
- Connor to develop main menu screens, etc
- We want to have genes with positives or drawbacks, need to consider this for ECS
- ECS options: GD Extensions ( core logic is in the GDE), or in godot with classic OO/inheritance
- Focus on PC/web as platform, don't worry about mobile (yet)
- John moving forward with GD Extensions implementation
- Use in-script docstrings for code documentation, use wiki for conceptual documentation
- Use and modify theme script (holds global constants for theme)
- Individually create slides and record, stitch videos together

**Comment by adediu25** [comment]
- Everyone review 3 group videos
- Game music work starting, using plant v zombies as inspo (more alien, less zombie)
- Connor dodge the codons mini game demo looking good
  - Change floating things to a nucleotide, catching correct 3 for amino acid builds it
  - Finish up and implement suggested changes
- John will move forward with trying rust for traits
- David figured out good way to use global variables, translation as well

**Comment by adediu25** [comment]
- Connor implemented changes to mini game, looks good!
- David worked on some changes and ideas to mutation minigame
- David will work on some art
- Connor polish up mini game and merge
- Alex start work on a punnett square video game

**Comment by adediu25** [comment]
- Connor finished up canges to dodge the codons game
- David showed off animated characters/enemies for the game
- we copied connor's minigame into develop branch and fixed dependency issues
- Final report is ready
- For next week maybe look into mockups for loading, main menu screens (not much due to thanksgiving)

**Comment by adediu25** [comment]
- Connor and alex post mini game demos on project summary doc
- John email advisor for review

**Comment by adediu25** [comment]
- merged all open PRs
- begin implementing testing framework for assignment
- plan to get at least a couple levels of main game complete for expo (need to complete trait system first)
- Alex to get pausing functionality working and start working on scene handler singleton

**Comment by adediu25** [comment]
- Set up gut addon for testing code
- started writing tests
- everyone write ~5 tests for their work
- fill out form for each test for the assigment

**Comment by IllustratedMan-code** [comment]
After Jan 16, we moved to an "as-needed" meeting style, so much of the "minutes" were instead recorded as issues, see the [repository issues](https://github.com/quaternary-game/quaternary/issues?q=is%3Aissue)
